### PR TITLE
Add own metric reporting to the example Server and Agent

### PR DESCRIPTION
**Pull request by tigrannajaryan**
The example server sends a connection settings to the agent
pointing to http://localhost:4318/v1/metrics

The example agent accepts the settings and sends a few of its
own metrics to the endpoint.

The results can be seen by running an OpenTelemtry Collector
with OTLP receiver on port 4318 and a for example using a
logging exporter for demo purposes.

Here is a sample output I get:

```
2022-01-10T18:59:46.610-0500	INFO	loggingexporter/logging_exporter.go:54	MetricsExporter	{"#metrics": 3}
2022-01-10T18:59:46.610-0500	DEBUG	loggingexporter/logging_exporter.go:64	ResourceMetrics #0
Resource labels:
     -> service.instance.id: STRING(01FS36A07R06AFVGQT5ZYC0GEK)
     -> service.name: STRING(io.opentelemetry.collector)
     -> service.version: STRING(1.0.0)
InstrumentationLibraryMetrics #0
InstrumentationLibrary opamp
Metric #0
Descriptor:
     -> Name: custom_metric_ticks
     -> Description:
     -> Unit:
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: AGGREGATION_TEMPORALITY_CUMULATIVE
NumberDataPoints #0
StartTimestamp: 2022-01-10 23:48:51.584797 +0000 UTC
Timestamp: 2022-01-10 23:59:46.610148 +0000 UTC
Value: 8515
Metric #1
Descriptor:
     -> Name: process.cpu.time
     -> Description:
     -> Unit:
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: AGGREGATION_TEMPORALITY_CUMULATIVE
NumberDataPoints #0
Data point attributes:
     -> state: STRING(user)
StartTimestamp: 2022-01-10 23:48:51.584797 +0000 UTC
Timestamp: 2022-01-10 23:59:46.610148 +0000 UTC
Value: 1.000000
NumberDataPoints #1
Data point attributes:
     -> state: STRING(system)
StartTimestamp: 2022-01-10 23:48:51.584797 +0000 UTC
Timestamp: 2022-01-10 23:59:46.610148 +0000 UTC
Value: 1.000000
NumberDataPoints #2
Data point attributes:
     -> state: STRING(wait)
StartTimestamp: 2022-01-10 23:48:51.584797 +0000 UTC
Timestamp: 2022-01-10 23:59:46.610148 +0000 UTC
Value: 0.159725
Metric #2
Descriptor:
     -> Name: process.memory.physical_usage
     -> Description:
     -> Unit:
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2022-01-10 23:59:46.610128 +0000 UTC
Value: 25072414
```

### CODE REVIEW

File: internal/examples/agent/agent/metricreporter.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	// Report the RSS, but also add some randomness to make it interesting for demo.
+	result.Observe(int64(memory.RSS) + rand.Int63n(10000000))
+}
+
+func (reporter *MetricReporter) sendMetrics() {
+
+	// Collect metrics every 5 seconds.
+	t := time.NewTicker(time.Second * 5)
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply internal/examples/agent/agent/metricreporter.go:177]: This is in the example agent, it is not part of the actual library.

[the remark this answers is not in the document]
Comment (pmm-sumo) [reply internal/examples/agent/agent/metricreporter.go:177]: Ahh, of course

**Review by djaglowski — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -7,18 +7,41 @@ require (
 	github.com/gorilla/websocket v1.4.2
 	github.com/knadh/koanf v1.3.3
 	github.com/oklog/ulid/v2 v2.0.2
+	github.com/shirou/gopsutil v3.21.11+incompatible
 	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/otel v1.3.0
```
Comment (Aneurysm9) [inline go.mod:12]: Might be good to have the example in a separate module to contain these dependencies.

**Review by Aneurysm9 — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (tigrannajaryan) [reply go.mod:12]: Yeah, I was thinking the same. We don't want the examples to pollute the dependencies of the library. I think I should introduce a separate module in a separate PR.

**Pull request merged**